src/backend/ticket_service/ticket_service.py

```diff
@@ -376,4 +377,253 @@ async def delete_ticket(ticket_id: str, customer_guid: str):
+
+
+@app.put("/tickets/{ticket_id}/comments/{comment_id}", response_model=CommentUpdateResponse, tags=["Comment Management"])
+async def update_comment(ticket_id: str, comment_id: str, comment_update: CommentUpdate, customer_guid: str):
+    """Update an existing comment"""
+    try:
+        update_status = db_manager.update_comment(ticket_id, comment_id, customer_guid, comment_update)
+
+        if update_status["status"] == "updated":
+            return CommentUpdateResponse(comment_id=comment_id, comment=comment_update.comment, is_edited=update_status["is_edited"], updated_at=update_status["updated_at"])
+        elif update_status["status"] == "not_found":
+            logger.error(update_status["reason"])
+            raise HTTPException(
+                status_code=HTTPStatus.NOT_FOUND,
+                detail=update_status["reason"]
+            )
+        elif update_status["status"] == "conflict":
+            original_error = update_status["reason"]
+            formatted_error = extract_core_error_details(original_error)
+            logger.error(f"Conflict error occurred:\n{formatted_error}")
+            raise HTTPException(
+                status_code=HTTPStatus.CONFLICT,
+                detail=formatted_error
+            )
+        elif update_status["status"] == "unknown_db":
+            logger.error("Unknown Database error occurred: " + update_status["reason"])
+            raise HTTPException(
+                status_code=HTTPStatus.BAD_REQUEST,
+                detail=update_status["reason"]
+            )
+        elif update_status["status"] == "bad_request":
+            original_error = update_status["reason"]
+            formatted_error = extract_core_error_details(original_error)
+            raise HTTPException(
+                status_code=HTTPStatus.BAD_REQUEST,
+                detail=formatted_error
+            )
+        elif update_status["status"] == "db_unreachable":
+            raise HTTPException(
+                status_code=HTTPStatus.SERVICE_UNAVAILABLE,
+                detail="The database is currently unreachable. Please try again later."
+            )
+        else:
+            logger.error(update_status["reason"])
+            raise HTTPException(
+                status_code=HTTPStatus.INTERNAL_SERVER_ERROR,
+                detail=f"Unexpected error: {update_status['reason'] or 'Unknown cause.'}"
+            )
+
+    except HTTPException as e:
+        raise e
+    except ValueError as e:
+        logger.error(f"Validation error: {e}")
+        raise HTTPException(status_code=HTTPStatus.BAD_REQUEST, detail=str(e))
+    except Exception as e:
+        logger.error(f"Unexpected error: {e}")
+        raise HTTPException(
+            status_code=HTTPStatus.INTERNAL_SERVER_ERROR,
+            detail="An unexpected server error occurred."
+        )
+
+
+@app.delete("/tickets/{ticket_id}/comments/{comment_id}", response_model=CommentResponse, tags=["Comment Management"])
```

Comment:
This API is again not intuitive
Make it /deletecomment

And then add the ticket id and comment_id in the parameters

Reply:
modified sir